test/ocamlorg_web/graphql_test.ml

```diff
@@ -0,0 +1,180 @@
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "0install-gtk")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "Decentralised installation system - GTK UI"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "abt")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "Package dependency solve"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "ocaml")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis = "OCaml port of CMU's abstract binding trees"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ; Package.create
+      ~name:(Package.Name.of_string "absolute")
+      ~version:(Package.Version.of_string "2.1.0")
+      { Package.Info.synopsis =
+          "Interactive theorem prover based on lambda-tree syntax"
+      ; description = ""
+      ; authors = []
+      ; maintainers = []
+      ; license = "ISC"
+      ; homepage = []
+      ; tags = []
+      ; dependencies = []
+      ; depopts = []
+      ; conflicts = []
+      ; url = None
+      }
+  ]
+
+let start_with_test str1 str2 cond () =
+  Alcotest.(check bool)
+    "is true or false"
+    cond
+    (Ocamlorg_web.Graphql.starts_with str1 str2)
+
+let is_package_test name1 name2 cond () =
+  Alcotest.(check bool)
+    "is false"
+    cond
+    (Ocamlorg_web.Graphql.is_package name1 name2)
+
+let get_packages_result_test () =
+  let total_packages = List.length packages in
+  let offset = 0 in
+  let limit = 3 in
+  let startswith = None in
+  let all_packages =
+    Ocamlorg_web.Graphql.get_packages_result
+      total_packages
+      offset
+      limit
+      startswith
+      packages
+  in
+  let num_of_pakgs = List.length all_packages.packages in
+  Alcotest.(check int) "returns 3 packages" limit num_of_pakgs
+
+let get_info_test () =
+  let deps = Ocamlorg_web.Graphql.get_info dependencies in
+  let num_of_deps = List.length deps in
+  Alcotest.(check int) "returns 6 dependencies" 6 num_of_deps
+
+let get_single_package_test name cond () =
```

Comment:
This is good but I think we also want to check that we got the correct package too (not just any package). To do this thoroughly we would need a `Package.t testable` (this is what `Alcotest.check` takes as an argument). The easiest way to do this is to add a "pretty printing" function to `package.ml` (This is because alcotest has `Alcotest.of_pp : 'a Fmt.t -> 'a Alcotest.testable`). Something along the lines of:

```ocaml
let pp : t Fmt.t = fun ppf t -> 
  Fmt.pf ppf {| { 
    name = %s; 
    version = %s;
    info = %a; 
  |} (Name.to_string t.name) (Version.to_string t.version) Info.pp (Lazy.force t.info)
```

Where you would also need `Info.pp`... Alternatively we can at least check the name and version of the returned package to make sure they are correct.

Reply:
I am a little bit confused on how to go about this please